app/domain/authentication/authn_gce/validate_resource_restrictions_configuration.rb

```diff
@@ -0,0 +1,49 @@
+module Authentication
+  module AuthnGce
+
+    # This class is responsible of validating resource restrictions configuration.
+    # the allowed configuration is: Role should have at least 1 permitted constraints, and non of illegal constraints
+    ValidateResourceRestrictionsConfiguration = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       %i(resource_restrictions permitted_constraints)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnGce::ValidatingResourceRestrictionsConfiguration.new)
```

Comment:
not a huge fan of this in the call since IMO call is a checklist, a quick summary of the code to follow but I can live with it

Reply:
i will add another question - do we even want this kind of log? we don't do it in other places. if we don't have something useful to log (e.g "working with provider uri {provider-uri}") then do we want to say that we are starting some procedure?

if the procedure will fail, an error will be raised. If not, why bother with this line?

again, i'd like to get @nessiLahav 's idea because he wrote this.

Reply:
I have a different opinion here, i really like that from reading of the logs you can identified the Big and small logic units. for example:
```
Authenticating host  
   Extracting resources
   .
   .
   Resources extracted
   Validating resources
   Resources are validated 
   .
   .
Host is authenticated

```

Also in cases of performance issues it will be more easy to identified where are the issues